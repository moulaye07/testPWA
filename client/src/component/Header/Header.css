div.Header {
  display: grid;
  grid-auto-flow: column;
  margin-top: 2%;
  grid-template-columns: 1fr 1fr;
  background-color: #db2f00;
  z-index: 1;
  width: 100vw;
}

a.userIcon {
  display: none;
}

a.navIcon {
  display: grid;
  justify-self: end;
  margin-right: 40%;
  height: 150%;
  background-color: #db2f00;
}

.userIcon:hover,
.navIcon:hover {
  opacity: 0.5;
}

.userIconA {
  display: grid;
  height: 100%;
  background-color: #db2f00;
}

.navIconA {
  display: grid;
  height: 100%;
  background-color: #db2f00;
}

.profile {
  flex: 1;
  display: flex;
  justify-content: flex-start;
  padding: 12px;
  background: white;
}

.menu {
  display: grid;
  justify-self: end;
  background-color: #db2f00;
}

.menuIn {
  display: none;
}

#lost {
  padding: 0;
  justify-self: end;
  margin-right: 10%;
  background-color: #db2f00;
}

@media (max-height: 812px) {
  div.Header {
    display: grid;
    grid-auto-flow: column;
    /* margin-top: 2%; */
    grid-template-columns: 1fr 1fr;
    background-color: #db2f00;
    height: 100%;
    width: 100vw;
  }

  a.userIcon {
    display: grid;
    margin-left: 10%;
    height: 100%;
    background-color: #db2f00;
  }

  .menuIn {
    display: grid;
    background-color: #db2f00;
  }

  a.navIcon {
    display: grid;
    justify-self: end;
    background-color: #db2f00;
  }

  .userIcon:hover,
  .navIcon:hover {
    opacity: 0.5;
  }

  .userIconA {
    display: grid;
    height: 100%;
    background-color: #db2f00;
  }

  .navIconA {
    display: grid;
    height: 100%;
    background-color: #db2f00;
  }
}

@media (max-height: 667px) {
}
