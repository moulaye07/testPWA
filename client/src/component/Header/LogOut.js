import React, { Component } from 'react';
import { logout } from '../../store/actions/loginActions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';


class LogOut extends Component {
  constructor(props) {
    super(props);
    this.onClick.bind(this)
  }


  onClick() {
    // this.props.logout()
    this.props.history.push('/')
  }

  render() {
    return (
      <div>
        <button onClick={this.props.logout()}>logout</button>
      </div>
    )
  }
}

LogOut.propTypes = {
  logout: PropTypes.func.isRequired
}


function mapStateToProps(state) {
  return {
    auth: state.auth,
    users: state.users
  }
}

export default connect(mapStateToProps, { logout })(LogOut);
