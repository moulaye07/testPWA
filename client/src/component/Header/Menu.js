import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import List from "@material-ui/core/List";
import DialogTitle from "@material-ui/core/DialogTitle";
import Dialog from "@material-ui/core/Dialog";
import MenuItem from "@material-ui/core/MenuItem";
import MenuIcon from "@material-ui/icons/Menu";
import IconButton from "@material-ui/core/IconButton";
import { NavLink } from "react-router-dom";
import { connect } from 'react-redux';

import "./Header.css";

const styles = {
  menuItem: {
    alignText: "center",
    color: "pink"
  }
};

class MenuShow extends React.Component {
  componentDidMount(props) {
  }

  constructor(props) {
    super(props);
    this.state = {
      selected: ""
      // isLoggedIn: false
    };
  }

  handleClose = () => {
    this.props.onClose(this.props.selectedValue);
  };

  handleListItemClick = value => {
    this.props.onClose(value);
  };

  render() {

    const { classes, onClose, isLoggedIn, ...other } = this.props;

    var menuStyle = {
      width: "250px",
      display: "flex",
      flexDirection: "column",
      alignItems: "center"
    };

    const style = {
      textAlign: "left",
      justifyContent: "center"
    };

    return (
      <Dialog
        onClose={this.handleClose}
        aria-labelledby="simple-dialog-title"
        {...other}
        max-width="md"
      >
        <div >
          <div style={menuStyle}>
            <DialogTitle id="simple-dialog-title">Menu</DialogTitle>
            {isLoggedIn==true ? (
              <List>
                <NavLink to="/">
                  <MenuItem
                    value="Home"
                    name="Home"
                    style={style}
                    onClick={() => this.handleListItemClick()}
                  >
                    Home
                  </MenuItem>
                </NavLink>
                <NavLink to="/post">
                  <MenuItem style={style} onClick={this.handleClose}>
                    profil
                  </MenuItem>
                </NavLink>
                <NavLink to="/">
                  <MenuItem style={style}  onClick={this.handleClose} >
                    logout
                  </MenuItem>
                </NavLink>
              </List>
            ) : (
              <List>
                <NavLink to="/">
                  <MenuItem
                    value="Home"
                    name="Home"
                    style={style}
                    onClick={() => this.handleListItemClick()}
                  >
                    Home
                  </MenuItem>
                </NavLink>
                <NavLink to="/post">
                  <MenuItem style={style} onClick={this.handleClose}>
                    Cities
                  </MenuItem>
                </NavLink>
                <NavLink to="/CreateAccount">
                  <MenuItem style={style} onClick={this.handleClose}>
                    Create Account
                  </MenuItem>
                </NavLink>
              </List>
            )}
          </div>
        </div>
      </Dialog>
    );
  }
}

MenuShow.propTypes = {
  classes: PropTypes.object.isRequired,
  onClose: PropTypes.func
};

const MenuWrapped = withStyles(styles)(MenuShow);

class Menu extends React.Component {
  state = {
    open: false
  };

  handleClickOpen = () => {
    this.setState({
      open: true,
      selected: ""
    });
  };

  handleClose = value => {
    this.setState({ open: false, selected: value });
  };

  render() {
    let isLoggedIn = this.props.login.isLoggedIn

    return (
      <div className="menuIn">
        <IconButton
          aria-label="More"
          aria-haspopup="true"
          onClick={this.handleClickOpen}
          id="lost"
        >
          <MenuIcon style={{ fontSize: 36, color: "#484848" }} />
        </IconButton>
        <MenuWrapped
          open={this.state.open}
          selected={this.state.selected}
          onClose={this.handleClose}
          isLoggedIn={isLoggedIn}
        />
      </div>
    );
  }
}


function mapStateToProps(state) {
  return {
    auth: state.auth,
    users: state.users,
    login: state.login
  }
}

export default connect(mapStateToProps)(Menu);
