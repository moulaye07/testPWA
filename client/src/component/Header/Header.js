/* eslint-disable no-unused-vars */
// eslint-disable-next-line no-unused-vars
import React, { Component } from 'react';
import './Header.css';
import { connect } from 'react-redux';
import Menu from './Menu';
import { NavLink } from 'react-router-dom';
import AccountCircle from '@material-ui/icons/AccountCircle';


class Header extends Component {

  render() {

    const isAuthenticated = this.props.login.isLoggedIn;
    const guestLink = (
      <div className="Header">
        <NavLink to="/login" className="userIcon">
          <AccountCircle style={{ fontSize: 36, color: '#484848' }} />
        </NavLink>
        <div className="navIcon" >
          <Menu />
        </div>
      </div>
    );

    const userLink = (
      <div className="Header">
        <NavLink to="/post" className="userIcon">
          <AccountCircle style={{ fontSize: 36, color: '#D8D8D8' }} />
        </NavLink>
        <div className="navIcon" >
          <Menu />
        </div>
      </div>
    );

    return (
      <>
        { isAuthenticated ? userLink : guestLink}
      </>
    );
  }
}

function mapStateToProps(state) {
  return {
    auth: state.auth,
    users: state.users,
    login: state.login
  }
}

export default connect(mapStateToProps)(Header);
