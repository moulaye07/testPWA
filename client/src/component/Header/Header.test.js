// eslint-disable-next-line no-unused-vars
import React from 'react';
import { shallow } from 'enzyme';
// eslint-disable-next-line no-unused-vars
import Header from './Header'
// import { wrap } from 'module';

const setUp = (props = {}) => {
  const component = shallow(<Header {...props} />)
  return component;
}

const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
}

describe('Header Component', () => {

  let component;
  beforeEach(() => {
    component = setUp()
  });

  it('should render-without errors', () => {
    console.log(component.debug());
    const wrapper = findByTestAtrr(component, Header)
    console.log(wrapper)

    expect(wrapper.length).toBe(0)
    expect(component).toMatchSnapshot();
    // expect(1).toBe(1);
  })
})
