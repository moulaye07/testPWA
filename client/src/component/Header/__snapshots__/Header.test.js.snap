// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render-without errors 1`] = `
<div
  className="Header"
>
  <a
    className="userIcon"
    href="/login"
  >
    <img
      alt="userIcon"
      className="userIconA"
      src="https://img.icons8.com/wired/64/000000/circled-user.png"
    />
  </a>
  <a
    className="navIcon"
    href="/"
  >
    <img
      alt="navIcon"
      className="navIconA"
      src="https://img.icons8.com/ios-glyphs/90/000000/menu.png"
    />
  </a>
</div>
`;
